psydac/linalg/kron.py

```diff
@@ -102,13 +102,23 @@ def dot( self, x, out=None ):
     #--------------------------------------
     # Other properties/methods
     #--------------------------------------
+    @property
+    def starts( self ):
+        return self._space.starts
+
+    # ...
+    @property
+    def ends( self ):
+        return self._space.ends
+
+    # ...
+    @property
+    def pads( self ):
+        return self._space.pads
 
+    # ...
     def __getitem__(self, key):
-        rows = key[:2]
-        cols = key[2:]
-        A = self._A1
-        B = self._A2
-        return A[rows[0],cols[0]]*B[rows[1],cols[1]]
+        raise NotImplementedError('TODO')
 
     # ...
     def tocoo( self ):
```

Comment:
We should remove `tocoo` and `tocsr` and just add the `tosparse` method instead. In linalg we only provide one sparse format of our choice, and we want to be able to change it whenever we want. The user is free to convert the sparse matrix to any other format.

Reply:
okay